engine/src/main/java/io/zeebe/engine/processor/workflow/CatchEventBehavior.java

```diff
@@ -286,4 +274,48 @@ private boolean sendOpenMessageSubscription(
 
     return extractedCorrelationKeys;
   }
+
+  protected static final class CorrelationKeyExtractionResultHandler
+      implements ExpressionResultHandler<String> {
+
+    private final MessageCorrelationKeyContext context;
+
+    protected CorrelationKeyExtractionResultHandler(MessageCorrelationKeyContext context) {
+      this.context = context;
+    }
+
+    @Override
+    public String handleEvaluationResult(final EvaluationResult evaluationResult) {
+      if (evaluationResult.isFailure()) {
+        throw new MessageCorrelationKeyException(context, evaluationResult.getFailureMessage());
+      }
+      if (evaluationResult.getType() == ResultType.STRING) {
+        return evaluationResult.getString();
+      } else if (evaluationResult.getType() == ResultType.NUMBER) {
+
+        final Number correlationKeyNumber = evaluationResult.getNumber();
+        if ((correlationKeyNumber instanceof Float) || (correlationKeyNumber instanceof Double)) {
```

Comment:
That's good you mention this.

My intention was to make sure that numbers with fractions don't get converted to Strings.
I cannot imagine a case in which a value of "42.5" is a valid correlation key.

The only case I could imagine it is if you have something like an article number with a dot in it (e.g. "178.452") and then for some reason it appears as a number rather than a string in the message. Still it is far-fetched.

I am slightly uncomfortable about the `getNumber().longValue` approach because `BigDecimal.of(54.5).longValue() == 54`, so I wouldn't catch the fraction and thus a potential developer error.

Reply:
will do as suggested